scripts/apps/archive/index.tsx

```diff
@@ -406,17 +417,83 @@ function spikeActivity(spike, data, modal, $location, $q, multi, privileges,
     _spike();
 
     function _spike() {
-        let message = gettext('Are you sure you want to spike the item?');
-
         if ($location.path() === '/workspace/personal') {
-            message = gettext('Do you want to delete the item permanently?');
-        } else if (get(privileges, 'privileges.planning') && data.item && data.item.assignment_id) {
-            message = gettext('This item is linked to in-progress planning coverage, spike anyway?');
-        } else if (!get(config, 'confirm_spike', true)) {
-            return spike.spike(data.item);
+            return modal.confirm(gettext('Do you want to delete the item permanently?'), gettext('Confirm'))
+                .then(() => spike.spike(data.item));
         }
 
-        return modal.confirm(message, gettext('Confirm'))
-            .then(() => spike.spike(data.item));
+        const onSpikeMiddlewares
+            : Array<IExtensionActivationResult['contributions']['entities']['article']['onSpike']>
+            = flatMap(
+                Object.values(extensions).map(({activationResult}) => activationResult),
+                (activationResult) =>
+                    activationResult.contributions != null
```

Comment:
Javascript needs optional chaining ASAP... It's on stage 2 right now https://github.com/tc39/proposal-optional-chaining 🤞🏻